File: lovable/src/config/nav.ts
/**
 * Navigation manifest for the AppShell.
 *
 * This is the single place to add/remove pages. In the future it should be
 * hydrated from the plugin registry (e.g. /api/plugins/routes) so the UI
 * only shows what is actually deployed.
 */
import {
  MessageSquare, BarChart3, Link2, FileText, Radio,
  Folder, Zap, Monitor, Settings, Bug, ScrollText, Shield,
  GitBranch, Orbit, Box, Globe, Network, Workflow, Brain,
} from "lucide-react";

export interface NavItem {
  title: string;
  path: string;
  icon: React.ComponentType<{ className?: string }>;
}

export interface NavGroup {
  label: string;
  items: NavItem[];
}

export const NAV_GROUPS: NavGroup[] = [
  {
    label: "Zeroclaw",
    items: [
      { title: "Antigravity Chat", path: "/chat", icon: MessageSquare },
      { title: "Routable Models", path: "/models", icon: Radio },
      { title: "LLM Gateway", path: "/llm", icon: Radio },
    ],
  },
  {
    label: "Control",
    items: [
      { title: "Overview", path: "/", icon: BarChart3 },
      { title: "Orchestration", path: "/orchestration", icon: Orbit },
      { title: "Services", path: "/services", icon: Link2 },
    ],
  },
  {
    label: "Agent",
    items: [
      { title: "Agents", path: "/agents", icon: Folder },
      { title: "Tools", path: "/tools", icon: Zap },
      { title: "Workflows", path: "/workflows", icon: GitBranch },
      { title: "Security", path: "/security", icon: Shield },
      { title: "Accountability", path: "/accountability", icon: ScrollText },
      { title: "Skills", path: "/skills", icon: ScrollText },
      { title: "Knowledge", path: "/knowledge", icon: Brain },
      { title: "Embedding Pipeline", path: "/embedding", icon: Orbit },
    ],
  },
  {
    label: "Infrastructure",
    items: [
      { title: "Containers", path: "/containers", icon: Box },
      { title: "Privacy Network", path: "/privacy-network", icon: Globe },
      { title: "Open vSwitch", path: "/ovs", icon: Network },
      { title: "OpenFlow", path: "/openflow", icon: Workflow },
      { title: "BTRFS Storage", path: "/btrfs", icon: Monitor },
      { title: "Data Stores", path: "/data-stores", icon: Monitor },
    ],
  },
  {
    label: "Settings",
    items: [
      { title: "Config", path: "/config", icon: Settings },
      { title: "Inspector", path: "/inspector", icon: Bug },
      { title: "State", path: "/state", icon: Monitor },
      { title: "Logs", path: "/logs", icon: ScrollText },
      { title: "gRPC Diagnostics", path: "/grpc", icon: Radio },
    ],
  },
];

File: crates/op-web/ui/src/nav.rs
//! Navigation model — 1:1 mirror of the React `NAV_GROUPS` array.

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
pub enum Route {
    // Console (top tab)
    Chat,
    Overview,
    Orchestration,
    Services,
    Sessions,
    Llm,
    Agents,
    Assistant,
    Models,
    Tools,
    Workflows,
    Security,
    Accountability,
    Skills,
    Knowledge,
    Embedding,
    Containers,
    PrivacyNetwork,
    Ovs,
    OpenFlow,
    Btrfs,
    DataStores,
    Config,
    Inspector,
    State,
    Logs,
    Grpc,
    GrpcExplorer,
    // Coming Soon (placeholders)
    Dreams,
    Reflections,
    Observability,
    LiveTrace,
    // Top-level tab
    Install,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
pub enum TopTab {
    Console,
    Install,
}

pub struct Item {
    pub title: &'static str,
    pub icon: &'static str,
    pub route: Route,
    pub placeholder: bool,
}

pub struct Group {
    pub label: &'static str,
    pub items: &'static [Item],
}

const fn it(title: &'static str, icon: &'static str, route: Route) -> Item {
    Item { title, icon, route, placeholder: false }
}
const fn placeholder(title: &'static str, icon: &'static str, route: Route) -> Item {
    Item { title, icon, route, placeholder: true }
}

pub static GROUPS: &[Group] = &[
    Group { label: "Chat", items: &[
        it("Chat", "💬", Route::Chat),
    ]},
    Group { label: "Control", items: &[
        it("Overview",      "▦", Route::Overview),
        it("Orchestration", "◎", Route::Orchestration),
        it("Services",      "⇄", Route::Services),
        it("Sessions",      "≡", Route::Sessions),
        it("LLM",           "◉", Route::Llm),
    ]},
    Group { label: "Agent", items: &[
        it("Agents",             "▤", Route::Agents),
        it("Assistant",          "✦", Route::Assistant),
        it("Routable Models",    "◉", Route::Models),
        it("Tools",              "⚡", Route::Tools),
        it("Workflows",          "⌥", Route::Workflows),
        it("Security",           "⛨", Route::Security),
        it("Accountability",     "✎", Route::Accountability),
        it("Skills",             "✎", Route::Skills),
        it("Knowledge",          "✷", Route::Knowledge),
        it("Embedding Pipeline", "◎", Route::Embedding),
    ]},
    Group { label: "Infrastructure", items: &[
        it("Containers",       "▣", Route::Containers),
        it("Privacy Network",  "⌖", Route::PrivacyNetwork),
        it("Open vSwitch",     "⌬", Route::Ovs),
        it("OpenFlow",         "⌘", Route::OpenFlow),
        it("BTRFS Storage",    "▥", Route::Btrfs),
        it("Data Stores",      "▥", Route::DataStores),
    ]},
    Group { label: "Settings", items: &[
        it("Config",           "⚙", Route::Config),
        it("Inspector",        "✱", Route::Inspector),
        it("State",            "▥", Route::State),
        it("Logs",             "✎", Route::Logs),
        it("gRPC Diagnostics", "◉", Route::Grpc),
        it("gRPC Explorer",    "⌁", Route::GrpcExplorer),
    ]},
    Group { label: "Coming Soon", items: &[
        placeholder("Dreams",        "☾", Route::Dreams),
        placeholder("Reflections",   "✦", Route::Reflections),
        placeholder("Observability", "◉", Route::Observability),
        placeholder("Live Trace",    "≈", Route::LiveTrace),
    ]},
];

pub fn route_title(r: Route) -> &'static str {
    for g in GROUPS {
        for i in g.items {
            if i.route == r { return i.title; }
        }
    }
    match r {
        Route::Install => "Install",
        _ => "ZeroClaw",
    }
}
